### PR TITLE
SPM-1729: Show only relevant OS filter options across tables

### PR DESCRIPTION
**Pull request by gkarat**
Implements https://issues.redhat.com/browse/SPM-1729.

This changes the behavior of the OS filter across tables. The filter now only shows the options that are relevant and available for the user's account. The options are requested from the Inventory API.

## How to test

1. Check the following pages and components to contain the same values for the OS filter:
1.1. https://stage.foo.redhat.com:1337/beta/insights/patch/systems
1.2. https://stage.foo.redhat.com:1337/beta/insights/patch/advisories/${advisory_id}
1.3. https://stage.foo.redhat.com:1337/beta/insights/patch/templates: create patch template wizard component.
2. Check whether a call to the Inventory API's /operating_system is made in all pages.
3. Check chips/filter selection behavior and URL parameters update.

## Screenshots

Screenshot 2022-09-15 at 15-27-15 Patch template - Patch Red Hat Insights
Screenshot 2022-09-15 at 15-26-53 FEDORA-EPEL-2022-929440e4dc - Advisories - Patch Red Hat Insights
Screenshot 2022-09-15 at 15-26-19 Systems - Patch Red Hat Insights

### CODE REVIEW

File: src/PresentationalComponents/Filters/OsVersionFilter.js [hunk 1]
```diff
@@ -1,71 +1,44 @@
...
+            scope: 'inventory',
+            module: './OsFilterHelpers'
         }
     );
+
+    let { os: currentValue } = currentFilter;
+    const osVersionValue = (
+        (typeof currentValue === 'string' && currentValue.split(',')) ||
+        []
+    ).map((version) => version.substr(5));
```
Comment (mkholjuraev) [inline src/PresentationalComponents/Filters/OsVersionFilter.js:19]: substr function is [deprecated](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr). Use [substring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring) instead.

**Review by mkholjuraev — CHANGES_REQUESTED** [review]
Overall code looks good. But, I would love to see a few duplicated code to be removed by making created OS filter hook more dynamic.  (I am checking how much effort is needed to use the default `operatingSystem` filter from [the Inventory table component](https://github.com/RedHatInsights/insights-inventory-frontend/blob/master/src/components/InventoryTable/EntityTableToolbar.js#L133))

File: src/SmartComponents/PatchSetWizard/steps/ReviewSystems.js [hunk 2]
```diff
@@ -2,7 +2,6 @@ import React, { useEffect, useState } from 'react';
 import propTypes from 'prop-types';
 import useFieldApi from '@data-driven-forms/react-form-renderer/use-field-api';
 import searchFilter from '../../../PresentationalComponents/Filters/SearchFilter';
-import osVersionFilter from '../../../PresentationalComponents/Filters/OsVersionFilter';
```
[the remark this answers is not in the document]
Comment (gkarat) [reply src/SmartComponents/PatchSetWizard/steps/ReviewSystems.js:5]: @mkholjuraev, thanks, this sounds reasonable. I have revisited this piece of code in the latest commit, PTAL

**Review by mkholjuraev — APPROVED** [review]
Looks perfect. @gkarat thank you! Did not run it though.

**Pull request merged**